src/OpenTelemetry.Instrumentation.GrpcNetClient/Implementation/GrpcClientDiagnosticListener.cs

```diff
@@ -135,16 +142,33 @@ public void OnStartActivity(Activity activity, object payload)
                 }
 
                 var uriHostNameType = Uri.CheckHostName(request.RequestUri.Host);
-                if (uriHostNameType == UriHostNameType.IPv4 || uriHostNameType == UriHostNameType.IPv6)
+                if (this.emitOldAttributes)
                 {
-                    activity.SetTag(SemanticConventions.AttributeNetPeerIp, request.RequestUri.Host);
+                    if (uriHostNameType == UriHostNameType.IPv4 || uriHostNameType == UriHostNameType.IPv6)
+                    {
+                        activity.SetTag(SemanticConventions.AttributeNetPeerIp, request.RequestUri.Host);
+                    }
+                    else
+                    {
+                        activity.SetTag(SemanticConventions.AttributeNetPeerName, request.RequestUri.Host);
+                    }
+
+                    activity.SetTag(SemanticConventions.AttributeNetPeerPort, request.RequestUri.Port);
                 }
-                else
+
+                if (this.emitNewAttributes)
```

Comment:
I would add a comment here linking to the spec you followed. This will help devs in the future understand where these attributes came from.